x/txfees/keeper/feedecorator.go

```diff
@@ -208,14 +208,14 @@ func (dfd DeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bo
 		return ctx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
-	// checks to make sure the module account has been set to collect fees in base token
-	if addr := dfd.ak.GetModuleAddress(types.FeeCollectorName); addr == nil {
-		return ctx, fmt.Errorf("fee collector module account (%s) has not been set", types.FeeCollectorName)
+	// checks to make sure the auth module account has been set to collect tx fees in base token, to be used for staking rewards
+	if addr := dfd.ak.GetModuleAddress(authtypes.FeeCollectorName); addr == nil {
+		return ctx, fmt.Errorf("fee collector module account (%s) has not been set", authtypes.FeeCollectorName)
 	}
 
-	// checks to make sure a separate module account has been set to collect fees not in base token
-	if addrNonNativeFee := dfd.ak.GetModuleAddress(types.FeeCollectorForStakingRewardsName); addrNonNativeFee == nil {
-		return ctx, fmt.Errorf("fee collector for staking module account (%s) has not been set", types.FeeCollectorForStakingRewardsName)
+	// checks to make sure a separate module account has been set to collect tx fees not in base token
+	if addrNonNativeFee := dfd.ak.GetModuleAddress(types.NonNativeTxFeeCollectorName); addrNonNativeFee == nil {
+		return ctx, fmt.Errorf("fee collector for staking module account (%s) has not been set", types.NonNativeTxFeeCollectorName)
```

Comment:
wtf this is in AnteHAndle. Lets make an issue to make this InitGenesis logic, were wasting gas on this every call

Reply:
oh its not an I/O call, its in the in-memory DB (module account permissions)